src/scenic/core/simulators.py

```diff
@@ -491,7 +501,7 @@ def updateObjects(self):
         """
         for obj in self.objects:
             # Get latest values of dynamic properties from simulation
-            dynTypes = obj._dynamicProperties
+            dynTypes = {prop: val for prop, val in obj._dynamicProperties.items() if prop not in obj._finalProperties}
```

Comment:
Let's not waste time computing this for every object every timestep. This is basically the only place `_dynamicProperties` is used, so I'd just set it up the way you want to use it when it is first created.